### PR TITLE
Update gulp optimize to rely on ejs for build dependency injection

### LINKED ISSUE #198: Update Production Build Process to rely soley on ejs
`index.html` presently relies on wiredep for dependency injection when building for production.  This action should be assumed, as it is in dev, by ejs or in this case we would use [gulp-ejs](https://www.npmjs.com/package/gulp-ejs) but not before all other dependencies are updated (to ensure seamless functionality)

### PR DESCRIPTION
**Pull request by AllenBW**
closes #198 

Removes a number of extraneous build tasks
Removes undesired `<--` comments from index
Improves build flow, personally I believe this flow to be more logical, easy to follow, readable

@chriskacerguis @himdel @jeff-phillips-18 @dtaylor113 FYSA nothing noticeably changes, except a few minor things (removing that silly comment `<--` comment n such)

### CODE REVIEW

File: client/index.ejs [hunk 1]
```diff
@@ -1,66 +1,44 @@
...
-  <!-- <%- include styles.html %> -->
+    <title ng-bind="title" translate>Self Service</title>
+    <meta charset="utf-8"/>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1"/>
+    <meta name="viewport"
+          content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"/>
+    <link rel="shortcut icon" href="/favicon.ico"/>
+    <link rel="icon" href="/favicon.ico"/>
+    <base href="/self_service/">
+    <style>
```
Comment (chriskacerguis) [inline client/index.ejs]: Is this style section needed?  I thought (I could be wrong) that it was in the CSS files?

**Comment by AllenBW** [comment]
@miq-bot add_label enhancement, euwe/no

File: client/index.ejs [hunk 2]
```diff
@@ -1,66 +1,44 @@
...
-
-  <!-- <%- include styles.html %> -->
+    <title ng-bind="title" translate>Self Service</title>
+    <meta charset="utf-8"/>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge, chrome=1"/>
+    <meta name="viewport"
+          content="width=device-width, initial-scale=1, maximum-scale=1, minimum-scale=1, user-scalable=no"/>
+    <link rel="shortcut icon" href="/favicon.ico"/>
+    <link rel="icon" href="/favicon.ico"/>
+    <base href="/self_service/">
```
Comment (chriskacerguis) [inline client/index.ejs]: @AllenBW Do we want that hard coded?

**Comment by himdel** [comment]
Well, :-1: for randomly changing the indent of `index.html`, but otherwise, wheee, finally no while line! :) :+1:

File: client/index.ejs (hunk 2 above)
Comment (AllenBW) [inline client/index.ejs]: @chriskacerguis not even sure we need this... will make a note to look at 🤘

**Comment by AllenBW** [comment]
`command + l` is my mistress, set at two spaces, she formats all my life!!!

hope **NOTHING** breaks (it shouldn't, this delivers identical end result)

**Pull request merged**